packages/@react-spectrum/s2/src/Table.tsx

```diff
@@ -11,85 +11,1103 @@
+
+interface S2TableProps {
+  /** Whether the Table should be displayed with a quiet style. */
+  isQuiet?: boolean,
+  /**
+   * Sets the amount of vertical padding within each cell.
+   * @default 'regular'
+   */
+  density?: 'compact' | 'spacious' | 'regular',
+  /**
+   * Sets the overflow behavior for the cell contents.
+   * @default 'truncate'
+   */
+  overflowMode?: 'wrap' | 'truncate',
+  // TODO: will we contine with onAction or rename to onRowAction like it is in RAC?
+  /** Handler that is called when a user performs an action on a row. */
+  onAction?: (key: Key) => void,
+  /**
+   * Handler that is called when a user starts a column resize.
+   */
+  onResizeStart?: (widths: Map<Key, ColumnSize>) => void,
+  /**
+   * Handler that is called when a user performs a column resize.
+   * Can be used with the width property on columns to put the column widths into
+   * a controlled state.
+   */
+  onResize?: (widths: Map<Key, ColumnSize>) => void,
+  /**
+   * Handler that is called after a user performs a column resize.
+   * Can be used to store the widths of columns for another future session.
+   */
+  onResizeEnd?: (widths: Map<Key, ColumnSize>) => void,
+  /** The current loading state of the table. */
+  loadingState?: LoadingState,
+  /** Handler that is called when more items should be loaded, e.g. while scrolling near the bottom. */
+  onLoadMore?: () => any
+}
+
+// TODO: Note that loadMore and loadingState are now on the Table instead of on the TableBody
+export interface TableProps extends Omit<RACTableProps, 'style' | 'disabledBehavior' | 'className' | 'onRowAction' | 'selectionBehavior' | 'onScroll' | 'onCellAction' | 'dragAndDropHooks'>, StyleProps, S2TableProps {
+}
+
+let InternalTableContext = createContext<TableProps & {scale?: Scale, layout?: S2TableLayout<unknown>, setIsInResizeMode?:(val: boolean) => void, isInResizeMode?: boolean}>({});
+
+const tableWrapper = style({
+  minHeight: 0,
+  minWidth: 0,
+  display: 'flex',
+  isolation: 'isolate',
+  disableTapHighlight: true
+});
+
+const table = style<TableRenderProps & S2TableProps & {scale?: Scale, isCheckboxSelection?: boolean}>({
+  width: 'full',
+  userSelect: 'none',
+  minHeight: 0,
+  minWidth: 0,
+  fontFamily: 'sans',
+  fontWeight: 'normal',
+  overflow: 'auto',
+  backgroundColor: {
+    default: 'gray-25',
+    isQuiet: 'transparent',
+    forcedColors: 'Background'
+  },
+  outlineColor: {
+    default: 'gray-300',
+    isFocusVisible: 'focus-ring',
+    forcedColors: 'ButtonBorder'
+  },
+  outlineWidth: {
+    default: 1,
+    isQuiet: 0,
+    isFocusVisible: 2
+  },
+  outlineStyle: 'solid',
+  borderRadius: {
+    default: size(6),
+    isQuiet: 'none'
+  },
+  // These scroll padding values are kinda odd (expected them to match the header height and checkbox column width) but they are the best fit from testing
+  scrollPaddingTop: {
+    scale: {
+      medium: '[21px]',
+      large: '[27px]'
```

Comment:
Wouldn't these need to be rems in case the user has a non-default font size set? Use `size` macro? But still weird that they don't match. Is it the border or something?

Reply:
I'm not quite sure why these values work to be honest, I did some digging here: https://github.com/adobe/react-spectrum/pull/6778#discussion_r1735343357 and got some varied results

Reply:
Filed https://issues.chromium.org/issues/365913982 against chrome and updated to the values that we expect to work. Core issue is that https://github.com/adobe/react-spectrum/blob/086ad3115c5f99115f623ddcd91a724599712cfd/packages/%40react-aria/utils/src/scrollIntoView.ts#L98 causes the scroll container to scroll even though the container is already in view